crates/strategies/blend-strategies/src/auction_manager.rs

```diff
@@ -225,6 +226,113 @@ impl OngoingAuction {
+                    requests.push(Request {
+                        request_type: 5,
+                        address: bid_asset.clone(),
+                        amount: *wallet_balance,
+                    });
+                    let wallet_dtoken_bal = wallet_balance
+                        .fixed_div_floor(SCALAR_9, reserve.est_d_rate)
+                        .unwrap();
+                    if wallet_dtoken_bal < bid_value {
+                        if new_pool_positions.liabilities.contains_key(&bid_asset) {
+                            new_pool_positions
+                                .liabilities
+                                .entry(bid_asset.clone())
+                                .and_modify(|e| *e += bid_value - wallet_dtoken_bal);
+                        } else {
+                            new_pool_positions
+                                .liabilities
+                                .insert(bid_asset.clone(), bid_value - wallet_dtoken_bal);
+                        }
+                    }
+                }
+            }
+        }
+        let mut liquidator_effective_collateral = sum_adj_asset_values(
+            new_pool_positions.collateral.clone(),
+            &self.pool,
+            true,
+            &self.db_manager,
+        )
+        .unwrap()
+        .1;
+        let liquidator_effective_liabilities = sum_adj_asset_values(
+            new_pool_positions.liabilities.clone(),
+            &self.pool,
+            false,
+            &self.db_manager,
+        )
+        .unwrap()
+        .1;
+        if self.auction_type == 0 {
+            for (index, asset) in supported_collateral.iter().enumerate() {
+                if index != 0 {
+                    let reserve = self
+                        .db_manager
+                        .get_reserve_config_from_asset(&self.pool, asset)?;
+                    let b_tokens = self.auction_data.lot.get(asset);
+                    if b_tokens.is_some() {
+                        if reserve.collateral_factor == 0
+                            || liquidator_effective_collateral
+                                > liquidator_effective_liabilities
+                                    .fixed_mul_ceil(*min_hf, SCALAR_7)
+                                    .unwrap()
+                        {
+                            requests.push(Request {
+                                request_type: 3,
+                                address: asset.clone(),
+                                amount: i64::MAX as i128,
+                            });
+                            liquidator_effective_collateral -= sum_adj_asset_values(
+                                HashMap::from([(asset.to_owned(), b_tokens.unwrap().to_owned())]),
+                                &self.pool,
+                                true,
+                                &self.db_manager,
+                            )
+                            .unwrap()
+                            .1
```

Comment:
The collateralized asset does not exist yet in the auctioneers effective collateral, so withdrawing it does not effect it